server/dash/dashboards/overview_combined.py

```diff
@@ -10,140 +10,240 @@
-    
+        group_by_col: the column name we use to groupby and output the result_counts_df dataframe.
+
     Returns: 
         result_df: the raw dataframe from calling the api with parameters from "api_params".
-        result_df_gb: dataframe output from result_df group by the column "group_by_col" and aggregating the sum.
+        result_counts_df: dataframe output from result_df group by the column "group_by_col" and aggregating the sum.
     """
     DATA_API_PATH = REPORT_API_PATH_ROOT + urllib.parse.urlencode(api_params)
     print(" * Downloading data from API path: " + DATA_API_PATH)
     result_df = pd.read_json(API_HOST + DATA_API_PATH)
-    result_df_gb = result_df.groupby([group_by_col])['counts'].sum().sort_values().to_frame()
+    result_counts_df = result_df.groupby([group_by_col])['counts'].sum().sort_values().to_frame()
     print(" * Dataframe has been loaded from API path: " + DATA_API_PATH)
-    return result_df, result_df_gb
+    return result_df, result_counts_df
 
-# DATA
+# DATAFRAMES
 
 # Loading the dataframe for the NCs and corresponding requests.
 nc_req_count_api_params = {'field': 'council_name', 'filter': CREATED_DATE_FILTER}
-_, nc_req_count_df = generate_dataframe_from_api(nc_req_count_api_params, nc_req_count_api_params['field'])
+_, nc_req_count_df = generate_dataframe_from_api(nc_req_count_api_params, COUNCIL_NAME_FIELD)
 
 # Loading the data for the number of new requests.
 new_req_count_api_params = {'field': 'created_year', 'filter': CREATED_DATE_FILTER}
-_, new_req_count_df = generate_dataframe_from_api(new_req_count_api_params, new_req_count_api_params['field'])
+_, new_req_count_df = generate_dataframe_from_api(new_req_count_api_params, CREATED_YEAR_FIELD)
 
 # Loading the count of each request types overall.
 req_count_api_params = {'field': 'type_name', 'filter': CREATED_DATE_FILTER}
-req_count_df, req_count = generate_dataframe_from_api(new_req_count_api_params, new_req_count_api_params['field'])
+req_count_df, req_count = generate_dataframe_from_api(req_count_api_params, TYPE_NAME_FIELD)
 
 # Loading the total number of request source.
 req_source_count_api_params = {'field': 'source_name', 'filter': CREATED_DATE_FILTER}
-req_source_count_df, req_source_count = generate_dataframe_from_api(req_source_count_api_params, req_source_count_api_params['field'])
+req_source_count_df, req_source_count = generate_dataframe_from_api(
+    req_source_count_api_params, SOURCE_NAME_FIELD)
 
 # Loading the number of Request Agencies.
 req_agency_count_api_params = {'field': 'agency_name', 'filter': CREATED_DATE_FILTER}
-req_agency_count_df, agency_count = generate_dataframe_from_api(req_agency_count_api_params, req_agency_count_api_params['field'])
+req_agency_count_df, agency_count = generate_dataframe_from_api(
+    req_agency_count_api_params, AGENCY_NAME_FIELD)
+
+# VISUALS HELPER FUNCTIONS.
+
+# Compute values for the indicator visuals.
+def get_counts_dict():
```

Comment:
I noticed that some of the variables produced by calls to `generate_dataframe_from_api` are only used within this function. Could we move those calls within this function?

That way, we know that the calls to `generate_dataframe_from_api` outside this function are shared variables, used by multiple things in this file.

Reply:
Thanks for the suggestion - resolved